### PR TITLE
Portable layer changes [Zynq, STM32, Pic32, ESP32, QEMU]

### PR DESCRIPTION
**Pull request by kar-rahul-aws**
Portable layer : Update Network Interface.c for EndPoint

Description
-----------
Files changed:
NetworkInterface.c for 
1. STM32Fxx
2. STM32Hxx
3. Pic32
4. DriverSAM
5. ESP32
6. MPS2_AN385 [QEMU]
7. Zynq

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/portable/NetworkInterface/DriverSAM/NetworkInterface.c [hunk 1]
```diff
@@ -136,6 +136,22 @@
 
 /*-----------------------------------------------------------*/
 
+/*
+ * A pointer to the network interface is needed later when receiving packets.
+ */
+static NetworkInterface_t * pxMyInterface = NULL;
+
+static BaseType_t xNetworkInterfaceInitialise( NetworkInterface_t * pxInterface );
```
Comment (htibosch) [inline source/portable/NetworkInterface/DriverSAM/NetworkInterface.c:144]: Would you mind the change the names of these these static function to "prvSAM_", like e.g. 
`prvSAM_NetworkInterfaceInitialise()` ?
In the labs/IPv6 branch you will find man y examples of this naming.

File: source/FreeRTOS_TCP_IP.c [hunk 2]
```diff
@@ -608,7 +608,7 @@
         configASSERT( pxNetworkBuffer != NULL );
         configASSERT( pxNetworkBuffer->pucEthernetBuffer != NULL );
 
-        BaseType_t xResult = pdPASS;
```
Comment (htibosch) [inline source/FreeRTOS_TCP_IP.c:611]: I would rather see `pdTRUE` and `pdFALSE` in stead. The values are the same bu the meaning is: "the network buffer has been released".

**Review by htibosch — COMMENTED** [review]
Thank you for all these changes. Please have a look at 2 minor remarks.

File: source/FreeRTOS_TCP_IP.c (hunk 2 above)
Comment (kar-rahul-aws) [reply source/FreeRTOS_TCP_IP.c:611]: Thanks for the review @htibosch . Updated .

**Pull request closed without merging**